packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,290 @@
+import { Inject, Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { SendTransactionOptions, WalletNotConnectedError, WalletNotReadyError } from '@solana/wallet-adapter-base';
+import { WalletName } from '@solana/wallet-adapter-wallets';
+import { Connection, Transaction } from '@solana/web3.js';
+import { asyncScheduler, combineLatest, defer, EMPTY, from, Observable, of, throwError } from 'rxjs';
+import { catchError, concatMap, filter, first, map, observeOn, switchMap, tap, withLatestFrom } from 'rxjs/operators';
+import { SignMessageNotFoundError } from '.';
+
+import { fromAdapterEvent, isNotNull } from '../operators';
+import {
+    SignAllTransactionsNotFoundError,
+    SignTransactionNotFoundError,
+    WalletNotSelectedError,
+} from './wallet.errors';
+import { WALLET_CONFIG } from './wallet.tokens';
+import { WalletConfig, WalletState } from './wallet.types';
+
+@Injectable()
+export class WalletStore extends ComponentStore<WalletState> {
+    private readonly _autoConnect = this._config?.autoConnect || false;
+    private readonly _localStorageKey = this._config?.localStorageKey || 'walletName';
+    readonly wallets$ = this.select((state) => state.wallets);
+    readonly selectedWallet$ = this.select((state) => state.selectedWallet);
+    readonly connected$ = this.select((state) => state.connected);
+    readonly connecting$ = this.select((state) => state.connecting);
+    readonly disconnecting$ = this.select((state) => state.disconnecting);
+    readonly wallet$ = this.select((state) => state.wallet);
+    readonly adapter$ = this.select((state) => state.adapter);
+    readonly publicKey$ = this.select((state) => state.publicKey);
+    readonly ready$ = this.select((state) => state.ready);
+    readonly anchorWallet$ = this.select(this.publicKey$.pipe(isNotNull), (publicKey) => ({
+        publicKey,
+        signTransaction: (transaction: Transaction) => this.signTransaction(transaction).toPromise(),
+        signAllTransactions: (transactions: Transaction[]) => this.signAllTransactions(transactions).toPromise(),
+    }));
```

Comment:
Anchor expects this interface: https://github.com/project-serum/anchor/blob/2604a442fda443f333cf7d3340892b95bc5f776b/ts/src/provider.ts#L220-L224

I'm thinking this should also feature detect if the signTransaction and signAllTransaction methods are defined. This way it's all or nothing, as in the React package.

Reply:
Gotcha